Affordable housing visualization draft version

ashwinmd opened the pull request:
Please review the first working version of the affordable housing visualization project.
The README has details on set up.

Commit:
Affordable housing visualization draft version

evankroske commented on affordable-housing/CSVToHTMLConverter.py:
```diff
@@ -0,0 +1,148 @@
+    
+
+def mapsUrlToHref(url, addressString):
+    retval = "<a href=\"" + url + " San Jose, CA\">" + addressString + "</a>"
```
", San Jose, CA, \">", right? Addresses look like this:

123 Jones Rd, Cityville, AA

ashwinmd replied:
I have fixed this with my latest commit. Please review

evankroske commented on affordable-housing/CSVToHTMLConverter.py:
```diff
@@ -0,0 +1,148 @@
+    return retval
+
+def javaScriptWriter():
+    retval = "<script>\r\nfunction loadDoc(streetName, street, streetType) {\r\n  var xhttp = new XMLHttpRequest();\r\n  \r\n\r\nvar settings = {\r\n  \"async\": true,\r\n  \"crossDomain\": true,\r\n  \"url\": \"http://www3.sanjoseca.gov/codeEnforcement/cets/New_ces_Results.asp\",\r\n  \"method\": \"POST\",\r\n  \"headers\": {\r\n    \"accept-language\": \"en-us\",\r\n    \"content-type\": \"appl ...
```
Could you move the JS to its own file so that I can review it line by line?

ashwinmd replied:
The html header and footer have been moved to their own files called housing_script_header.txt and housing_script_footer.txt.
I did not create a separate js file but the code is now in a readable form in housing_script_header.txt.

evankroske commented on affordable-housing/CSVToHTMLConverter.py:
```diff
@@ -0,0 +1,148 @@
+    return retval
+    
+
+if len(sys.argv) < 3:
```
All the code at the top level (not function definitions) should be wrapped in "if **name** == "**main**:". Example:

http://www.diveintopython.net/getting_to_know_python/index.html#odbchelper.divein

ashwinmd replied:
I have added this wrapper line.

evankroske commented on affordable-housing/CSVToHTMLConverter.py:
```diff
@@ -0,0 +1,148 @@
+# initialize rownum variable
+rownum = 0
+
+htmlfile.write("<!DOCTYPE html>\r\n<script type=\"text/javascript\" src=\"http://code.jquery.com/jquery-1.7.1.min.js\"></script>\r\n<html>\r\n<body>\r\n\r\n<h2>AFFORDABLE PROPERTIES IN SAN JOSE</h2>")
```
Instead of mixing Python an HTML, could you create a template file in HTML, read it, and expand it using string.Template?

https://docs.python.org/2/library/string.html#template-strings

This would make the code more readable and easier for me to review.

ashwinmd replied:
I have used string templates in this latest version.

evankroske commented on affordable-housing/LICENSE:
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
```
You can remove this file. There's a license file at the top level of this repo.

ashwinmd replied:
Done

evankroske commented on affordable-housing/PublishAffordableExcelCSV.csv:
```diff
@@ -0,0 +1,231 @@
+Coming Soon,,,,,
```
Was this copied from the open data portal?

http://data.sanjoseca.gov/datasets/165367/affordable-housing-data/

If so, could you delete it? Instead, provide instructions or a script for downloading it.

ashwinmd replied:
I have updated the README with the download instructions.

evankroske commented on affordable-housing/housing.html:
```diff
@@ -0,0 +1,901 @@
+<!DOCTYPE html>
```
This file should go into a separate pull request to the gh-pages branch. Please put it in a new directory here and link to it from index.html:

https://github.com/codeforsanjose/HackNightProjects/tree/gh-pages

ashwinmd replied:
I created the housing.html and housing.kml files only to illustrate the sample outputs. I am not sure if you want me to move both of them to the gh-pages branch.

Commit:
KML generator to create a kml document featuring off leash dog areas from mountainview.gov

Commit:
Revert "KML generator to create a kml document featuring off leash dog areas from mountainview.gov"
Per code review, this should be in a different pull request

This reverts commit 6e60c7483bf3d6efb81c39f5d56c1e7c818c7a35.

Commit:
Code Review feedback incorporated -

1 - Added templates for applicable html strings
2 - Deleted the license file
3 - Created separate files to store scripts and html headers and footers
4 - Added instructions on downloading the dataset

New features
5 - Added code to generate KML document with affordable housing information

Commit:
KML generator to create a kml document featuring off leash dog areas from mountainview.gov

ashwinmd commented:
Evan, I tried reverting the offleash dog park KML generator and adding it back but it automatically is picked up by the pull request.

evankroske commented on affordable-housing/CSVToHTMLConverter.py:
```diff
@@ -0,0 +1,191 @@
+def templateGenerator(cells):
+    row = '<tr>'
+    for cell in cells:
+        s = Template('<td>$column</td>')
```
Sorry to change my mind, but Template is overkill for this situation. String,format would be more concise.

ashwinmd replied:
No problem , I have incorporated this feedback

evankroske commented on affordable-housing/CSVToHTMLConverter.py:
```diff
@@ -0,0 +1,191 @@
+    return row
+
+def codeEnforcerLinkGenerator(streetNum, street, streetType):
+    s = Template("<a href=javascript:loadDoc(\'$num\',\'$name\',\'$type\')>Link</a>")
```
Also overkill.

ashwinmd replied:
I have deleted this and similar pieces of redundant code

evankroske commented on affordable-housing/CSVToHTMLConverter.py:
```diff
@@ -0,0 +1,191 @@
+    s = Template("<a href=javascript:loadDoc(\'$num\',\'$name\',\'$type\')>Link</a>")
+    return s.substitute(num = streetNum, name = street, type = streetType)
+
+def loadHTMLHeader(htmlfile):
```
Instead of creating a file for the header and another for the footer, you should create one file with both with a template variable for the contents in between. This would be a good use of string.Template.

ashwinmd replied:
Done, please review. The new file is called housing_script.txt

evankroske commented on affordable-housing/CSVToHTMLConverter.py:
```diff
@@ -0,0 +1,191 @@
+        for column in row:
+            if column != '':
+                columnList.append(column)
+            if columnNum == 1:
```
This will always be true.

ashwinmd replied:
Have reworked this

evankroske commented on affordable-housing/CSVToHTMLConverter.py:
```diff
@@ -0,0 +1,191 @@
+    kmlFile.close()
+       
+
+    exit(0)
```
Unnecessary

evankroske commented on affordable-housing/CSVToHTMLConverter.py:
```diff
@@ -0,0 +1,191 @@
+   
+    print sys.argv
+    
+    kmlDoc = xml.dom.minidom.Document()
```
Could you create a new script for creating the KML document from the affordable housing data? It's confusing that this script creates a KML document and an web page.

ashwinmd replied:
Done, the new script is called CSVToKML.py. I have updated the README with relevant instructions

Commit:
Incorporate more code review feedback
- Separated the html and kml generation into different scripts
- Merged the html header and footer into a single file
- Used a string template to insert the generated housing table into the html file
- Added wait time to prevent exceeding the google map query limit
- Updated README and sample outputs

evankroske commented:
Could you create a new pull request for the branch gh-pages with the table of affordable housing units and their code enforcement cases?
